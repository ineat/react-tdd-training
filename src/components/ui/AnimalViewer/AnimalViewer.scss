.animal-viewer {
  display: flex;
  flex-direction: column;
  align-items: center;

  .animal-details {
    display: flex;
    width: 50%;
    flex-direction: column;

    .picture, .description {
      width: 100%;
      border: 1px solid black;
    }

    .description {
      font-size: 20px;
      padding: 2rem 0;
    }
  }

  button {
    width: 200px;
    height: 50px;
    margin: 1rem;
    border-radius: 5px;
    font-size: 14px;
    background-color: lightgreen;
    color: white;
    font-weight: bold;
    cursor: pointer;
    transition: background-color 0.5s;

    &:hover {
      background-color: lightseagreen;
    }
  }
}