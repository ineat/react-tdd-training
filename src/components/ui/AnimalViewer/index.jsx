import * as React from 'react';

import './AnimalViewer.scss';

export default ({ img, description, onReloadImage }) => (
  <div className="animal-viewer">
    <div className="animal-details">
      <img src={img} className="picture" alt="Moohhh"/>
      <div className="description">{description || 'No description'}</div>
    </div>
    <button className="reload-button" onClick={onReloadImage}>I want another one !</button>
  </div>
);