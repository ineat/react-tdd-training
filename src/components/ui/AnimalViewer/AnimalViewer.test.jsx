import * as React from 'react';
import { mount, shallow } from 'enzyme';

import AnimalViewer from '.';

describe('Animal viewer', () => {
  const defaultProps = {
    img: 'my-dog.jpg',
    description: 'Nice animal',
    onReloadImage: jest.fn()
  }
  it('Should render.', () => {
    expect(mount(<AnimalViewer { ...defaultProps } />));
  });

  it('Should call "onReloadImage" prop when clicking on button.', () => {
    const wrapper = shallow(<AnimalViewer { ...defaultProps } />);

    wrapper.find('.reload-button').prop('onClick')();

    expect(defaultProps.onReloadImage).toHaveBeenCalled();
  });

  it('Should display "No description" when no description are provided.', () => {
    const props = {
      ...defaultProps,
      description: null
    };
    const wrapper = shallow(<AnimalViewer { ...props } />);

    expect(wrapper.find('.description').text()).toEqual('No description');
  });

  it('Should display the description when the description is provided.', () => {
    const wrapper = shallow(<AnimalViewer { ...defaultProps } />);

    expect(wrapper.find('.description').text()).toEqual(defaultProps.description);
  });
});