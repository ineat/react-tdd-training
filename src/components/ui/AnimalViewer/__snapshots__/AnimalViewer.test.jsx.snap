// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal viewer Should render. 1`] = `
<_default
  description="Nice animal"
  img="my-dog.jpg"
  onReloadImage={[MockFunction]}
>
  <div
    className="animal-viewer"
  >
    <div
      className="animal-details"
    >
      <img
        alt="Moohhh"
        className="picture"
        src="my-dog.jpg"
      />
      <div
        className="description"
      >
        Nice animal
      </div>
    </div>
    <button
      className="reload-button"
      onClick={[MockFunction]}
    >
      I want another one !
    </button>
  </div>
</_default>
`;
